lib/core/context.c

```diff
@@ -150,6 +150,17 @@ void h2o_context_dispose(h2o_context_t *ctx)
     h2o_filecache_destroy(ctx->filecache);
     ctx->filecache = NULL;
 
+    /* clear storage */
+    for (i = 0; i != ctx->storage.size; ++i) {
+        h2o_context_storage_item_t item = ctx->storage.entries[i];
```

Comment:
This is kind of a nitpick but maybe you would want to do `h2o_context_storage_item_t *item = ctx->storage.entries + i` to avoid copy.